wgpu-core/src/device/resource.rs

```diff
@@ -1902,7 +1916,19 @@ impl<A: HalApi> Device<A> {
             return Err(DeviceError::WrongDevice.into());
         }
 
-        check_buffer_usage(bb.buffer_id, buffer.usage, pub_usage)?;
+        check_buffer_usage(
+            bb.buffer_id,
+            buffer.usage
+                // Allow indirect buffers to be used as storage buffers, allowing
+                // us to bind them and validate their content. Note that we already
+                // pass this usage to hal at buffer creation.
+                | if buffer.usage.contains(wgt::BufferUsages::INDIRECT) {
+                    wgt::BufferUsages::STORAGE
+                } else {
+                    wgt::BufferUsages::empty()
+                },
```

Comment:
Do we need to be passing these into check_buffer_usage? If the user will do something wrong with usage, they will get an error message that includes a buffer usage that they did not request. We should just assume our code is right (or strict_assert maybe).

Reply:
Good point!